client/my-sites/checkout/composite-checkout/use-prepare-product-for-cart.js

```diff
@@ -168,29 +164,32 @@ function useAddProductFromSlug( {
 	const isFetchingProducts = useSelector( ( state ) => isProductsListFetching( state ) );
 	const products = useSelector( ( state ) => getProductsList( state ) );
 
-	// Special handling for search products: Always add Jetpack Search to
-	// Jetpack sites and WPCOM Search to WordPress.com sites, regardless of
-	// which slug was provided. This allows e.g. code on jetpack.com to
-	// redirect to a valid checkout URL for a search purchase without worrying
-	// about which type of site the user has.
-	if ( productAlias && JETPACK_SEARCH_PRODUCTS.includes( productAlias ) ) {
-		if ( isJetpackNotAtomic ) {
-			productAlias = productAlias.includes( 'monthly' )
-				? PRODUCT_JETPACK_SEARCH_MONTHLY
-				: PRODUCT_JETPACK_SEARCH;
-		} else {
-			productAlias = productAlias.includes( 'monthly' )
-				? PRODUCT_WPCOM_SEARCH_MONTHLY
-				: PRODUCT_WPCOM_SEARCH;
-		}
-	}
-
-	const product = useSelector( ( state ) =>
-		getProductBySlug( state, getProductSlugFromAlias( productAlias ) )
+	// If `productAliasFromUrl` has a comma ',' in it, we will assume it's because it's
+	// referencing more than one product. Because of this, the rest of this function will
+	// work with an array of products even if `productAliasFromUrl` includes only one.
+	const validProducts = useMemo(
+		() =>
+			productAliasFromUrl &&
+			productAliasFromUrl
+				.split( ',' )
+				// Special treatment for Jetpack Search products
+				.map( ( productAlias ) => getJetpackSearchForSite( productAlias, isJetpackNotAtomic ) )
+				// Get the product information if it exists, and keep a reference to its product alias
+				.map( ( productAlias ) => {
+					const validProduct = products[ getProductSlugFromAlias( productAlias ) ];
+					return validProduct ? { ...validProduct, product_alias: productAlias } : validProduct;
```

Comment:
We add the `productAlias` to the product object since we will need it later when we create the cart item.

Reply:
FWIW, most of the times `productAlias` will be equal to a real product slug. Exceptions are themes, domains, and a concierge session.